sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusSessionProcessorOptions.cs

```diff
@@ -0,0 +1,175 @@
+        /// <value>The maximum number of concurrent calls to the callback.</value>
+        public int MaxConcurrentCalls
+        {
+            get => _maxConcurrentCalls;
+
+            set
+            {
+                Argument.AssertAtLeast(value, 1, nameof(MaxConcurrentCalls));
+                _maxConcurrentCalls = value;
+            }
+        }
+        private int _maxConcurrentCalls;
+
+        /// <summary>
+        /// An optional list of session IDs to scope
+        /// the <see cref="ServiceBusSessionProcessor"/> to. If left
+        /// blank, the processor will not be limited to any specific
+        /// session IDs.
+        /// </summary>
+        public string[] SessionIds { get; set; }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" /> is equal to this instance.
+        /// </summary>
+        ///
+        /// <param name="obj">The <see cref="System.Object" /> to compare with this instance.</param>
+        ///
+        /// <returns><c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.</returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override bool Equals(object obj) => base.Equals(obj);
+
+        /// <summary>
+        /// Returns a hash code for this instance.
+        /// </summary>
+        ///
+        /// <returns>A hash code for this instance, suitable for use in hashing algorithms and data structures like a hash table.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        /// Converts the instance to string representation.
+        /// </summary>
+        ///
+        /// <returns>A <see cref="System.String" /> that represents this instance.</returns>
+        ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public override string ToString() => base.ToString();
+
+        /// <summary>
+        /// Creates a new copy of the current <see cref="ServiceBusProcessorOptions" />, cloning its attributes into a new instance.
+        /// </summary>
+        ///
+        /// <returns>A new copy of <see cref="ServiceBusProcessorOptions" />.</returns>
+        internal ServiceBusProcessorOptions Clone()
+        {
+            var clone = new ServiceBusProcessorOptions
+            {
+                ReceiveMode = ReceiveMode,
+                PrefetchCount = PrefetchCount,
+                AutoComplete = AutoComplete,
+                MaxAutoLockRenewalDuration = MaxAutoLockRenewalDuration,
+                MaxReceiveWaitTime = MaxReceiveWaitTime
+            };
+            if (MaxConcurrentCalls > 0)
+            {
+                clone.MaxConcurrentCalls = MaxConcurrentCalls;
+            }
+            return clone;
```

Comment:
I think this should be `ServiceBusSessionProcessorOptions`.

Reply:
No, this is to translate between session and regular processor options.